File: java/frame/mine.java
class tile {
	private int value;			// tile value
	private boolean checked;	// tile checked
	private boolean flaged;		// tile flaged
	
	tile(int v){
		value = v;			// set value
		checked = false;	// defualt
		flaged = false;		// defualt
	}
	
	public int getVal(){ // return tile value
		return this.value;
	}
	
	public boolean getCheck(){ // return tile checked
		return this.checked;
	}
	
	public boolean getflag(){ // return tile flaged
		return this.flaged;
	}
}

class map {
	int x;			// Columns
	int y;			// Rows
	tile[][] Map;	// Tile map
	
	map(){ // no input method
		x = 7;					// defualt size
		y = 7;					// defualt size
		Map = new tile[y][x];	// defualt size
		
		// for every row and column in the Map
		for(int i = 0; i < y; i++){
			for(int j = 0; j < x; j++){
				// Set tiles
				Map[i][j] = new tile(0);
			} 
		}
	}
	
	map(int ix, int iy){ // custom input method
		if(ix > 0){ x = ix; }else{ x = 7; }	// valid size or defualt
		if(iy > 0){ y = iy; }else{ y = 7; } // valid size or defualt
		Map = new tile[iy][ix];				// custom map size
		
		// for every row and column in the Map
		for(int i = 0; i < iy; i++){
			for(int j = 0; j < ix; j++){
				// Set tiles
				Map[i][j] = new tile(0);
			} 
		}
	}
	
	public void drawBoard(){
		String ret = "";	// create blank string
		
		// for every row and column in the Map
		for(int i = 0; i < this.y; i++){
			for(int j = 0; j < this.x; j++){
				// add value to string
				ret = ret + this.Map[i][j].getVal() + ",";
			}
			
			// newline
			ret = ret + "\n";
		}
		
		// print string
		System.out.println(ret);
	}
	
	public boolean checkTile(int x,int y){ // Check tile at (x,y)
		// check if value is bomb
		if(this.Map[y][x].getVal() < 0){ return false; }
		return true;
	}
	
	public boolean mapClear(){ // Check if the map is clear
		// placeholder method
		return true;
	}
}

public class mine{ // Public Class
	
	public static void main(String[] args){ // Starter Function
		map M = new map(); // create map
		System.out.println("x:" + M.x + ", y:" + M.y);
		M.drawBoard(); // Draw The Board
		// Zero Errors
	}
}

File: lua/frame/mine.lua
local x -- Columns
local y -- Rows

local map = {} -- create maps
x = 7
y = 7

-- init map by setting it blank
for i = 1,y do
	map[i] = map[i] or {}
	for j = 1,x do
		map[i][j] = { -- tile object
			value = 0;			-- default tile value
			checked = false;	-- default tile checked
			flaged = false;		-- default tile flaged
		}
	end
end

function drawBoard() -- draw the map
	str = "" -- blank string
	
	-- for each row and column
	for i = 1,y do
		for j = 1,x do
			-- append the value of the tile
			str = str..map[i][j].value..","
		end
		
		-- newline
		str = str.."\n"
	end
	
	-- print to screen
	print(str)
end

function getPlayerInput(input)end

function checkTile(tile) -- check if tile is a bomb
	-- is tile less than 0
	if(tile.value < 0) then
		return true
	end
	return false
end

function mapClear()
	-- place holder function
	return true
end

print("x:"..x..", y:"..y)
drawBoard()

return 0; -- Zero Errors
